processors/drop_event.go

```diff
@@ -34,31 +34,29 @@ func (f *EventDropper) Init(options map[string]interface{}) error {
 	}
 	f.config = config
 
-	values := make(map[string]bool)
+	f.values = make(map[string]struct{}, len(f.config.Values))
 	for _, val := range f.config.Values {
-		values[val] = true
+		f.values[val] = struct{}{}
 	}
-	f.values = values
 	return nil
 }
 
 func (f *EventDropper) Process(ev *event.Event) bool {
-	if ev.Data != nil {
-		val, ok := ev.Data[f.config.Field]
-		if !ok {
-			return true
-		}
-		valString, ok := val.(string)
-		if !ok {
-			logrus.WithFields(logrus.Fields{
-				"key":   f.config.Field,
-				"value": val}).
-				Debug("Not filtering field of non-string type")
-		}
-		_, exists := f.values[valString]
-		if exists {
-			return false
-		}
+	if ev.Data == nil {
+		return true
+	}
+	val, ok := ev.Data[f.config.Field]
+	if !ok {
+		return true
+	}
+	valString, ok := val.(string)
+	if !ok {
+		logrus.WithFields(logrus.Fields{
+			"key":   f.config.Field,
+			"value": val,
+			"type":  fmt.Sprintf("%T", val)}).
+			Debug("Not filtering field of non-string type")
```

Comment:
Did you mean to return false here?

Reply:
I think it should return true, as it clearly doesn't match the explicit list of terms we should drop.